### ISSUE TITLE
[ENH][MNT] umbrella issue: interface estimators (other than `AutoARIMA`) from `statsforecast` into `sktime`

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#4643]
This issue lists different tasks to interface more estimators from `statsforecast`. Till `0.18.0`, `AutoARIMA` (present in `statsforecast.arima`) was already interfaced as `StatsForecastAutoARIMA` inheriting more `ARIMA` focused base adapter `_StatsForecastAdapter`. As part of #4629, a more general base class has been attempted to conform to the general pattern of `statsforecast.models.*` estimators, and `StatsForecastAutoARIMA` has been updated accordingly. Given new general adapter, one can now add other `statsforecast` estimators inheriting from `_GeneralisedStatsForecastAdapter`.

Estimators to add:

1. `Automatic Foreacsting` models (more popular)
  1. [`AutoETS`](https://nixtla.github.io/statsforecast/models.html#autoets)
  2. [`AutoCES`](https://nixtla.github.io/statsforecast/models.html#autoces)
  3. [`AutoTheta`](https://nixtla.github.io/statsforecast/models.html#autotheta) - #4539
2. Other models listed [here](https://nixtla.github.io/statsforecast/models.html)

Expected tasks for each PR to add one of the models:

1. create a new concrete estimator class inheriting from `_GeneralisedStatsForecastAdapter`
2. add appropriate tags, especially `ignores-exogeneous-X`, `capability:pred_int` and `capability:pred_int:insample`
3. add all positional and keyword arguments required by the underlying estimator in `__init__` method, instead of using `*args` or `**kwargs`
4. create an instance of underlying `statsforecast` model in `_instantiate_model` method (indicated as abstact class in adapter)
5. add at least one, and preferably more (but not too high to reduce CI time), test cases in `get_test_params` method

To start with, this model specific classes can live in `sktime/forecasting/statsforecast.py`. Please keep @FedericoGarza as the author of the underlying estimators. The new class should be exported in `__all__` and also should be mentioned in `docs/source/api_reference/forecasting.rst`.

After (most of) these models are added, a maintainenace PR can split into different model into individual modules inside a directory (sub-package) of same name as current module (i.e. `sktime/forecasting/statsforecast/`) to help maintenance and avoid deprecation lag.

**Comment by yarnabrina** [comment]
Based on discussions in Discord on 2023-05-27, this can be done by adding new estimator as suggested above, or it can also be used through a composition class. In that case, it will be user's responsibility to create an instace of any `statsforecast` model and pass it through a delegator class to conform to `sktime` contracts.

The main challenge in this case will be to assign appropriate tags to allow/restrict several `sktime` functionalities, or any other way to achieve the same. On the other hand, the advantage will be that it will not be required to create a wrapper class for every single model.

(@fkiraly: please edit if I misinterpreted/miscommunicated any of your suggestions.)

**Comment by fkiraly** [comment]
Yes, the 2 possible patterns here are: adapter parent class, and adapter composition/wrapper class.
Just to summarize what I said in the design discussion:

Pattern no.1 is as in the linked PR; pattern no.2 would create a class `StatsforecastToSktime` or similar, this can then be used as `sktime_est = StatsforecastToSktime(sf_est, tags={"capability:pred_int", etc})` etc, i.e., it also attaches the tags.

No.2 is harder to use for an everyday user, but it is robust to additions of new estimators to `statsforecast`, while no.1 requires developer action for every new estimator.

Now, there is a way to have your cake and eat it - create `StatsforecastToSktime`, but use it only as a developer tool, together with the `_DelegatedForecaster`. Although that might not be too different from just doing pattern no.1.

I think you made a couple good arguments that doing no.2 with the `_DelegatedForecaster` might be more work to provide all `statsforecast` forecasters in an `sktime` interface, because the `_GeneralizedStatsforecastAdapter` already gives the developer a relatively easy (albeit repetitive) way to add 1:1 interfaces, that is not that different from the no.2 plus delegation option. It's basically adding the tags which in `statsforecast` only figure in the main readme.

FYI @FedericoGarza in case you have sth to add.

**Comment by fkiraly** [comment]
There is actually a fourth option - use pattern no.1 for 1:1 interfacing, and also create the `StatsforecastToSktime` adapter to cover estimators without an 1:1 interface, e.g., future estimators added to `statsforecast`. Although that might be least favourable in terms of work overhead.

**Comment by fkiraly** [comment]
@yarnabrina, my virus scanner claims that `nixtla.github.io` is blacklisted - are we aware of any hack or security vulnerabilities?

**Comment by fkiraly** [comment]
> That link does not work by the way, you have to use https://nixtla.github.io/statsforecast/.

The link I used is verbatim the one you posted under point 2 at the top, by clicking on "here"

**Comment by yarnabrina** [comment]
At least 2 weird thing is going on here:

1. The link in "here" of point 2 is opening `https://nixtla.github.io/statsforecast/models.html` for me, have no idea why is it different for you.
2. The above url still fails, even though I'm certain it was working before. Nixtla probably updated their site structure, now the working link is `https://nixtla.github.io/statsforecast/src/core/models.html`.

Fixed the link above to avoid confusions in future.

**Comment by yarnabrina** [comment]
Adding `AutoTBATS` as part of #5908.

**Comment by fkiraly** [comment]
which ones are we still missing? I suppose intermittent demand and hierarchical?

**Comment by yarnabrina** [comment]
These are all as of v1.7.2.

```pycon
>>> from statsforecast import models
>>> models.__all__
['AutoARIMA', 'AutoETS', 'ETS', 'AutoCES', 'AutoTheta', 'ARIMA', 'AutoRegressive', 'SimpleExponentialSmoothing', 'SimpleExponentialSmoothingOptimized', 'SeasonalExponentialSmoothing', 'SeasonalExponentialSmoothingOptimized', 'Holt', 'HoltWinters', 'HistoricAverage', 'Naive', 'RandomWalkWithDrift', 'SeasonalNaive', 'WindowAverage', 'SeasonalWindowAverage', 'ADIDA', 'CrostonClassic', 'CrostonOptimized', 'CrostonSBA', 'IMAPA', 'TSB', 'MSTL', 'TBATS', 'AutoTBATS', 'Theta', 'OptimizedTheta', 'DynamicTheta', 'DynamicOptimizedTheta', 'GARCH', 'ARCH', 'ConstantModel', 'ZeroModel', 'NaNModel']
```

I only added 4 Auto-* ones, AutoARIMA we had already. I think ARCH, GARCH were contributed as well, but they are kept in a different sub-directory (another iteration of directory structure conflict), and we have MSTL back adapter.

We don't have interface to rest, but probably have most of them in some other shape or form.

**Comment by fkiraly** [comment]
That's a couple of obscure/esoteric ones - ADIDA, IMAPA, what even are these

Anyway, a checklist at the top would be useful for new contributors?

**Comment by fkiraly** [comment]
@tblume1992, re MFLES, your assistance would be greatly appreciated! If you want, you can of course own the estimator!

**Comment by tblume1992** [comment]
@fkiraly yeah definitely let me know if you need anything! Unfortunately I won't have the time to do the PR but I can definitely support any code snippets or parameter questions!
